plugins/system/ps/disk/disk_darwin.go

```diff
@@ -74,7 +79,10 @@ func DiskPartitions(all bool) ([]DiskPartitionStat, error) {
 			Fstype:     common.IntToString(stat.Fstypename[:]),
 			Opts:       opts,
 		}
-		ret = append(ret, d)
+		_, ok := set[d.Fstype]
+		if ok || len(fstypes) == 0 {
```

Comment:
I'm new to Go, and don't know if this would be more idiomatic:

    if _, ok := set[d.Fstype]; ok || len(fstypes) == 0 {